api/envoy/api/v2/core/config_source.proto

```diff
@@ -15,9 +15,24 @@ option java_multiple_files = true;
 
 // [#protodoc-title: Configuration sources]
 
+// xDS API version. This is used to describe both resource and transport
+// protocol versions (in distinct configuration fields).
+enum ApiVersion {
+  // When not specified, we assume v2, to ease migration to Envoy's stable API
+  // versioning. If a client does not support v2 (e.g. due to deprecation), this
+  // is an invalid value.
+  AUTO = 0;
+
+  // Use xDS v2 API.
+  V2 = 1;
+
+  // Use xDS v3alpha API.
+  V3ALPHA = 2;
+}
```

Comment:
We are going to accumulate deprecated enum values for every new version we have, and immediately have one for V3ALPHA -> V3. Did you consider using a string here instead? I could see pros/cons to both. Internally we can map the string to an enum/etc. to make it much easier to fail invalid config, deprecate, etc. WDYT?

Reply:
Since we can deprecate enum values, this only have the trailing 2 versions (plus alpha variants), so I think this is reasonably bounded. My intuition would be that a string is too weakly typed given the limited number of options here.

Reply:
OK sounds good.